spec/migrations/20190729170013_split_storages_per_ems_spec.rb

```diff
@@ -0,0 +1,163 @@
+        end
+      end
+
+      context "and two active hosts" do
+        let(:ems) { ems_stub.create! }
+        let!(:host_1) do
+          host_stub.create!(:ems_id => ems.id).tap do |h|
+            host_storage_stub.create!(:host_id => h.id, :storage_id => storage.id)
+          end
+        end
+        let!(:host_2) do
+          host_stub.create!(:ems_id => ems.id).tap do |h|
+            host_storage_stub.create!(:host_id => h.id, :storage_id => storage.id)
+          end
+        end
+
+        it "doesn't touch the storage record" do
+          migrate
+          expect(storage_stub.count).to eq(1)
+        end
+
+        it "both hosts are still linked to the storage" do
+          migrate
+
+          storage = storage_stub.first
+          expect(host_1.reload.storages.first).to eq(storage)
+          expect(host_2.reload.storages.first).to eq(storage)
+        end
+      end
+
+      context "with different datastores in the same EMS" do
+        let(:ems) { ems_stub.create! }
+        let!(:host_1) do
+          host_stub.create!(:ems_id => ems.id).tap do |h|
+            host_storage_stub.create!(:host_id => h.id, :storage_id => storage.id, :ems_ref => "datastore-1")
+          end
+        end
+        let!(:host_2) do
+          host_stub.create!(:ems_id => ems.id).tap do |h|
+            host_storage_stub.create!(:host_id => h.id, :storage_id => storage.id, :ems_ref => "datastore-2")
+          end
+        end
+
+        it "creates two storages" do
+          migrate
+          expect(storage_stub.count).to eq(2)
+        end
+
+        it "links the right host to the right datastore" do
+          migrate
+          expect(host_1.reload.storages.first.ems_ref).to eq("datastore-1")
+          expect(host_2.reload.storages.first.ems_ref).to eq("datastore-2")
+        end
+      end
+
+      context "with different datastores in different EMSs" do
+        let(:ems_1) { ems_stub.create! }
+        let(:ems_2) { ems_stub.create! }
+        let!(:host_1) do
+          host_stub.create!(:ems_id => ems_1.id).tap do |h|
+            host_storage_stub.create!(:host_id => h.id, :storage_id => storage.id, :ems_ref => "datastore-1")
+          end
+        end
+        let!(:host_2) do
+          host_stub.create!(:ems_id => ems_2.id).tap do |h|
+            host_storage_stub.create!(:host_id => h.id, :storage_id => storage.id, :ems_ref => "datastore-2")
+          end
+        end
+
+        it "creates two storages" do
+          migrate
+          expect(storage_stub.count).to eq(2)
+          expect(storage_stub.pluck(:ems_id)).to match_array([ems_1.id, ems_2.id])
+        end
+
+        it "links the right host to the right datastore" do
+          migrate
+
+          expect(host_1.reload.storages.first.ems_id).to eq(host_1.ems_id)
+          expect(host_2.reload.storages.first.ems_id).to eq(host_2.ems_id)
+        end
+
+        context "with an archived host" do
+          let(:archived_host) do
```

Comment:
It's missing `!`, the host never gets created..

And then, the syntax error, as `ems_ref` is undefined :)

Reply:
Yes, see https://github.com/ManageIQ/manageiq-schema/pull/444